### ISSUE TITLE
tf.math.tanh can produce values outside of `[-1, 1]`

### ISSUE THREAD

**Issue by jburnim** [tensorflow/tensorflow#35435]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):  yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  https://colab.research.google.com/
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.1.0-rc2
- Python version: 3.6.9

**Describe the current behavior**

```
import numpy as np
import tensorflow as tf

x = np.float32(-8.51089)
print(tf.math.tanh(x))  # => <tf.Tensor: shape=(), dtype=float32, numpy=-1.0000001>
print(tf.math.tanh(-x))  # => <tf.Tensor: shape=(), dtype=float32, numpy=1.0000001>
```

**Describe the expected behavior**

The output of `tanh(x)` should always be `>= -1` and `<= 1`.

**Code to reproduce the issue**

The above code is a minimal repro.

You can also reproduce the issue using the following colab: https://colab.research.google.com/drive/1BHGZoBgMzOlG0e3cYhGYEnLPp1_uSpIY .

**Other info / logs**

This issue was originally filed as https://github.com/tensorflow/tensorflow/issues/34773 against TF 1.15.0 in Colab, but it is not clear if the issue could be reproduced in TF 1.15.0.

I haven't done an exhaustive search, but it looks like this issue exists in TF 2.1.0-dev20191203 and TF 2.1.0-dev20191218, as well, and was fixed in TF 2.1.0-dev20191219.

 - TF 2.1.0-dev20191218 repro: https://colab.research.google.com/drive/1GfA45tP2Q1ZQDmRIusfz4mmfTA5TW1ne

 - Correct output in TF 2.1.0-dev20191219: https://colab.research.google.com/drive/104qs8RAzWmsHATfCZ-01NbnBDnhqRzv0

If the issue has been fixed in TF 2.1.0-dev20191219,  does this mean the issue will be fixed in TF 2.1.0?

**Comment by gadagashwini** [comment]
@jburnim, Its fixed in latest Tf-nightly (2.1.0.dev20191226). It will be fixed in Future release 2.1.
Please find the [gist](https://colab.sandbox.google.com/gist/gadagashwini/3516941d5de628801730adeb5e632b56/untitled328.ipynb). Thanks!

**Comment by jburnim** [comment]
I just tested in the newly-released TF 2.1.0, and I'm still getting an incorrect result:
```
>>> import numpy as np
>>> import tensorflow as tf
>>> tf.__version__
'2.1.0'
>>> print(tf.math.tanh(x))
tf.Tensor(-1.0000001, shape=(), dtype=float32)
>>> print(tf.math.tanh(-x))
tf.Tensor(1.0000001, shape=(), dtype=float32)

**Comment by jvishnuvardhan** [comment]
@jburnim Thanks for reporting this issue. I was able to reproduce the issue in `TF2.1`. However, I cannot reproduce the issue with `tf-nightly`. I am sure this was resolved with `tf-nightly`.

I am closing the issue as it was resolved in `tf-nightly`, but feel free to reopen the issue if it persists again. Thanks!
